### PR TITLE
Enforce scalastyle

### PR DESCRIPTION
**Pull request by mmartina**
- trigger scalastyle sbt plugin as part of compile and test configuration
- revised and cleaned up scalastyle config
- added/enabled more rules, as specified in:
  http://www.scalastyle.org/rules-1.0.0.html
- fixed scalastyle issues or disabled specific rules on the spot
- introduced String helper function `toOneLine` to work with multiline text
- deprecated Dim.=<(value: String)

### CODE REVIEW

**Comment by Fokko** [comment]
Should we remove scalafmt then?

**Comment by mmartina** [comment]
> Should we remove scalafmt then?

I do not think so. Scalafmt and Scalastyle are complementary tools.
Yes, in part they control the same aspects and can be contradictory. Therefore, they both need to be carefully fine-tuned.

**Comment by barend** [comment]
Regarding the `sbt-scalafmt`, I noticed that the com.lucidchart plugin we're using is on low freq maintenance (i.e. "done"), but there's another plugin of the same name but different groupId that's much more recently maintained.

I chose to stick with the one we're already using to minimize impact. If we're doing a larger overhaul in this area, we might want to update to the newer one.

File: scalastyle-config.xml [hunk 1]
```diff
@@ -67,29 +66,35 @@ This file is divided into 3 sections:
...
-            <parameter name="ignoreImports">true</parameter>
+            <parameter name="tabSize"><![CDATA[4]]></parameter>
+            <parameter name="ignoreImports"><![CDATA[false]]></parameter>
         </parameters>
     </check>
 
     <check level="error" class="org.scalastyle.scalariform.ClassNamesChecker" enabled="true">
         <parameters>
-            <parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter>
+            <parameter name="regex"><![CDATA[^[A-Z][A-Za-z]*$]]></parameter>
```
Comment (barend) [inline scalastyle-config.xml:85]: Scala allows underscores, number characters and non-latin characters. I don't have a concrete example, but I can see valid use for adding `0-9` to the second group in the pattern.

**Comment by barend** [comment]
LGTM

**Comment by anskarl** [comment]
LGTM too

File: scalastyle-config.xml (hunk 1 above)
Comment (mmartina) [reply scalastyle-config.xml:85]: `^[A-Z][A-Za-z]*$` is just scalastyle's default value.
in case needed in future, we can make the pattern more lenient.

**Pull request merged**